Framework/Core/include/Framework/AnalysisHelpers.h

```diff
@@ -482,6 +482,8 @@ auto getTableFromFilter(const T& table, soa::SelectionVector&& selection)
   }
 }
 
+void initializePartitionCaches(std::set<uint32_t> const& hashes, std::shared_ptr<arrow::Schema> const& schema, expressions::Filter const& filter, gandiva::NodePtr& tree, gandiva::FilterPtr& gfilter);
```

Comment:
I would use a sorted `std::vector<uint32_t>` in place of a std::set, given you create / sort it only once.

Reply:
I agree, however it should be done in a separate PR, it needs some other changes.